### PR TITLE
Feature/351 refactor person component with BaseFormComponent

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: frontend/app/components/base-form-component.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default class BaseFormComponent extends Component {
       .then(savedRecords => {
         this.handleSubmitSuccessful(savedRecords);
         this.afterSubmit();
+        return true;
```
Comment (njaeggi) [inline frontend/app/components/base-form-component.js:35]: why was this added?, just curious

File: frontend/app/components/person-form.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    let create = this.record.id;
+    super.submit(person).then(
+      r =>
+        r &&
+        super
+          .submit(attributes)
+          .then(r => r && this.savePicture(picturePath))
+          .then(r => r && this.afterSuccess(create))
+    );
+  }
```
Comment (njaeggi) [inline frontend/app/components/person-form.js:36]: what does prettier say here :grin:

File: frontend/app/components/person-form.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    personRole.personRoleLevel = personRoleLevel;
+  }
+
+  @action
+  setNationality(country) {
+    this.record.nationality = country[0];
+  }
+
+  @action
+  setNationality2(country) {
```
Comment (njaeggi) [inline frontend/app/components/person-form.js:126]: don't use magic numbers, maybe name this action something like "setSecondNationality"

File: frontend/app/components/picture-edit.js [hunk 4]
```diff
@@ -1,67 +1,36 @@
...
+  constructor() {
+    super(...arguments);
+    this.picturePath = this.args.imagePath;
+  }
+  @action
+  uploadImage(e) {
+    let file = e.target.files[0];
     if (!/\.(?:jpe?g|png|gif|svg|bmp)$/i.test(file.name)) {
-      this.get("notify").alert("Invalider Datentyp");
+      this.notify.alert("Invalider Datentyp");
```
Comment (njaeggi) [inline frontend/app/components/picture-edit.js:19]: add translation

File: frontend/app/components/picture-edit.js [hunk 5]
```diff
@@ -1,67 +1,36 @@
...
       return;
     }
 
     if (file.size > 10000000) {
       //10MB
-      this.get("notify").alert("Datei ist zu gross, max 10MB");
+      this.notify.alert("Datei ist zu gross, max 10MB");
```
Comment (njaeggi) [inline frontend/app/components/picture-edit.js:25]: add translation

File: frontend/app/components/picture-edit.js [hunk 6]
```diff
@@ -1,67 +1,36 @@
...
-
-  didInsertElement() {
-    this.$(".img-input").on("change", e => {
-      if (e.target.files.length) {
-        this.uploadImage(e.target.files[0]);
-        e.target.value = null;
-      }
-    });
+    this.picturePath = URL.createObjectURL(file);
+    this.args.person.picturePath = URL.createObjectURL(file);
```
Comment (njaeggi) [inline frontend/app/components/picture-edit.js:29]: nice work! :rocket:

File: frontend/app/models/person.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ const Person = DS.Model.extend({
   birthdate: DS.attr("date"),
   picturePath: DS.attr("string"),
   location: DS.attr("string"),
-  maritalStatus: DS.attr("string"),
+  maritalStatus: DS.attr("string", { defaultValue: "single" }),
```
Comment (njaeggi) [inline frontend/app/models/person.js:9]: is this really needed?, since it wasn't needed before, or was this a bug?

File: frontend/tests/acceptance/create-person-test.js [hunk 8]
```diff
@@ -2,14 +2,20 @@ import { module, test, skip } from "qunit";
...
+import { click, currentURL } from "@ember/test-helpers";
 import setupApplicationTest from "frontend/tests/helpers/setup-application-test";
-import { currentURL } from "@ember/test-helpers";
 import { selectChoose } from "ember-power-select/test-support";
+import Notify from "ember-notify";
 
 module("Acceptance | create person", function(hooks) {
   setupApplicationTest(hooks);
 
+  const notifyStub = Notify.extend({
```
Comment (njaeggi) [inline frontend/tests/acceptance/create-person-test.js:13]: really nice! :1st_place_medal:

File: frontend/tests/acceptance/edit-person-test.js [hunk 9]
```diff
@@ -11,7 +11,7 @@ import { fillIn } from "@ember/test-helpers";
 module("Acceptance | edit person", function(hooks) {
   setupApplicationTest(hooks);
 
-  test("/people/:id edit person data", async function(assert) {
+  skip("/people/:id edit person data", async function(assert) {
```
Comment (njaeggi) [inline frontend/tests/acceptance/edit-person-test.js:14]: why do we skip this now, is this because of the pikadate bug?

**Review by njaeggi — DISMISSED** [review]

File: frontend/app/components/person-form.js (hunk 3 above)
Comment (kcinay055679) [reply frontend/app/components/person-form.js:126]: That wasn't me who introduced this name

File: frontend/app/components/person-form.js (hunk 2 above)
Comment (njaeggi) [reply frontend/app/components/person-form.js:36]: I would add another line between the function, seems weird that prettier says this is fine :thinking:

File: frontend/app/models/person.js (hunk 7 above)
Comment (kcinay055679) [reply frontend/app/models/person.js:9]: This is necessary now

File: frontend/app/components/person-form.js (hunk 2 above)
Comment (kcinay055679) [reply frontend/app/components/person-form.js:36]: Prettier wants it exactly like this

File: frontend/tests/acceptance/edit-person-test.js (hunk 9 above)
Comment (Robin481) [reply frontend/tests/acceptance/edit-person-test.js:14]: Then add a comment why it is skipped :wink:

File: frontend/app/models/person.js (hunk 7 above)
Comment (Robin481) [reply frontend/app/models/person.js:9]: Maybe elaborate a bit don't just say yes? :grin:

Comment (kcinay055679) [reply frontend/app/models/person.js:9]: It's easier and more readable to set the maritalStaus default value right in the model itself.
Is was defined in the logic of the selector previously.

**Comment by kcinay055679** [comment]
After the rebase on master two tests failed, i tried to fix them but due to a bug in the pikaday addon i could not solve the bug, or at least i think thats the source of the error. Then i tried to simply skip those tests but then ember hit with me with the exception that i have to call setupApplicationContext before i can call visit(). But first i don't call visit() anywhere and second, even if i called setupApplicationContext the errors still appears.

File: frontend/config/environment.js [hunk 10]
```diff
@@ -54,7 +54,7 @@ module.exports = function(environment) {
   if (environment === "test") {
     // Testem prefers this...
     ENV.locationType = "none";
-
+    ENV.APP.autoboot = false;
```
Comment (Robin481) [inline frontend/config/environment.js:57]: Is this really needed? I am assuming it fixed your `same root element` error?
what happens if you stop splitting the person-form test into 2 different modules?

File: frontend/tests/acceptance/create-person-test.js [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+module("Acceptance | create person", function(hooks) {
+  setupApplicationTest(hooks);
+
+  const notifyStub = Notify.extend({
+    alert(message, options) {
+      options.closeAfter = null;
+      return this.show("alert", message, options);
+    }
+  });
+  /// Skip this test since there is a bug wtih the @renderInPlace attribute  of the powerselect
```
Comment (Robin481) [inline frontend/tests/acceptance/create-person-test.js:19]: Did you create some sort of follow-up for this?

File: frontend/tests/acceptance/create-person-test.js [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+      "Email muss ausgefüllt werden"
+    );
+    assert.equal(
+      document.querySelectorAll(".ember-notify")[1].querySelector(".message")
+        .innerText,
+      "Email Format nicht gültig"
+    );
+  });
+
+  // Skip this test since there is a bug wtih the @renderInPlace attribute  of the powerselect
```
Comment (Robin481) [inline frontend/tests/acceptance/create-person-test.js:125]: Same here. follow up?

File: frontend/tests/pages/people-new.js [hunk 13]
```diff
@@ -12,16 +12,17 @@ const { resolve } = RSVP;
...
     toggleNationalitiesCheckbox: clickable("#toggle-nationalities-id"),
+    newRoleButton: clickable("[data-test-new-function]"),
 
     toggleNewForm() {
       return this.newForm;
     },
 
     async createPerson(person) {
-      await Object.keys(person).reduce(
+      Object.keys(person).reduce(
```
Comment (Robin481) [inline frontend/tests/pages/people-new.js:25]: Does this not cause a problem if it isn't `await` anymore?

**Review by Robin481 — CHANGES_REQUESTED** [review]
Nice work. A few notes and this:

- The translations for `person-role` errors are missing (create an empty role and press save).
- The errors for `person-role` only show up if it is the only error (fill in everything but Wohnsitz and create an empty Role -> press Save -> it will only show the Wohnsitz error -> Fill in a Wohnsitz and only then will it show the `person-role` errors).
- Translations for Language Skills are missing as well and the error above also occurs for the languages (create a new empty language skill and press save). 
- When creating a profile from another profile view and pressing save you will be redirected to the previously visited profile instead of the newly created one. (go to /people/1 -> New profile -> Save -> you are back to /people/1 instead of /people/newProfileId

File: frontend/tests/acceptance/create-person-test.js (hunk 12 above)
Comment (kcinay055679) [reply frontend/tests/acceptance/create-person-test.js:125]: https://github.com/orgs/puzzle/projects/5/views/1?pane=issue&itemId=27710056

File: frontend/tests/acceptance/create-person-test.js (hunk 11 above)
Comment (kcinay055679) [reply frontend/tests/acceptance/create-person-test.js:19]: https://github.com/orgs/puzzle/projects/5/views/1?pane=issue&itemId=27710056

**Review by Robin481 — CHANGES_REQUESTED** [review]
Hello.
Looking way better, but I figured the error out that I wasn't able to reproduce when we checked together.

New Profile -> Add a picture -> Fill out everything -> Add a role and press save -> you will get the role-errors -> Fix them and save -> The User will be created with the default picture instead of the one uploaded :grin:

**Review by Robin481 — APPROVED** [review]
:rocket:
